modules/roles.js

```diff
@@ -0,0 +1,202 @@
+    if (inv.length == 0) int.reply({ content: "You don't have any colors in your inventory!", ephemeral: true });
+    else int.reply({ embeds: [embed], ephemeral: !(int.channel?.id == int.client.getTextChannel(u.sf.channels.botspam)?.id) });
+  } catch (e) { u.errorHandler(e, int); }
+}
+
+/** @param {Augur.GuildInteraction<"CommandSlash">} int */
+async function equip(int) {
+  try {
+    int.deferReply({ ephemeral: true });
+    const allColors = eRoles.map(r => r.id);
+    const available = getInventory(int.member);
+    const input = int.options.getString("color");
+    if (!input) {
+      await int.member.roles.remove(allColors);
+      return int.editReply("Color removed!");
+    }
+    const real = int.guild.roles.cache.find(r => r.name.toLowerCase() == input.toLowerCase());
+    const role = int.guild.roles.cache.find(r => r.id == eRoles.find(a => a.baseRole == real?.id || a.id == real?.id)?.id);
+    if (!role) {
+      return int.editReply("Sorry, that's not a color role on this server. Check `/role inventory` to see what you can equip.");
+    } else if (!available.has(role.id)) {
+      return int.editReply("Sorry, you don't have that color in your inventory. Check `/role inventory` to see what you can equip.");
+    } else {
+      await int.member.roles.remove(allColors);
+      await int.member.roles.add(role.id);
+      int.editReply("Color applied!");
+    }
+  } catch (e) { u.errorHandler(e, int); }
+}
+
+const Module = new Augur.Module()
+  .addInteraction({
+    name: "role",
+    guildId: u.sf.ldsg,
+    onlyGuild: true,
+    id: u.sf.commands.slashRole,
+    process: async (interaction) => {
+      if (interaction.isAutocomplete()) return;
+      switch (interaction.options.getSubcommand(true)) {
+      case "add": return await roleFunc(interaction);
+      case "remove": return await roleFunc(interaction, false);
+      case "give": return staffFunc(interaction);
+      case "take": return staffFunc(interaction, false);
+      case "inventory": return inventory(interaction);
+      case "equip": return equip(interaction);
+      case "whohas": return whoHas(interaction);
+      }
+    },
+    autocomplete: (interaction) => {
+      const option = interaction.options.getFocused(true);
+      if (option.name == 'role') {
+        if (interaction.options.getSubcommand(true) == "whohas") {
+          const all = Module.client.guilds.cache.get(u.sf.ldsg).roles.cache.filter((n) => n?.name.toLowerCase().startsWith(option.value?.toLowerCase())).map((n) => n.name).slice(0, 24);
+          return interaction.respond(u.unique(all).map(v => ({ name: v, value: v })));
+        } else if (interaction.options.getSubcommand(true) == "give" || interaction.options.getSubcommand(true) == "take") {
+          if (!u.perms.calc(interaction.member, ["team", "mod", "mgr"])) return;
+          const values = ["Adulting", "Bookworm", "LDSG Lady"].filter(n => n?.toLowerCase().startsWith(option.value?.toLowerCase())).slice(0, 24);
```

Comment:
I don't think you'll need to filter or slice on this one since it's only 3 options